### DISCUSSION TITLE
Is there a way to use leptos without wasm?

### DISCUSSION THREAD

**Discussion by nripendra** [leptos-rs/leptos#2892] [General]
I'm trying to use leptos as purely a template engine. I really like breaking apps into manageable components. However, I couldn't find anyway to not generate any wasm at all. Is it even possible?

I have tried to find this information in tutorials, issues and discussions cannot seem to fine anything.

**Comment by gbj** [comment]
Yes. Are you maybe confusing using `leptos` the crate with using `cargo-leptos` the build tool? Just adding `leptos` as a dependency and using a function like [`render_to_string`](https://docs.rs/leptos/latest/leptos/ssr/fn.render_to_string.html) (or, in 0.7/git main, some view `.to_html()`) will give you an HTML string you can use in any context. (Activate the `experimental-islands` and `ssr` features for cleanest results here)

**Reply by nripendra** [reply]
Thanks for a quick reply. Is there any example on how to configure app to use just the core leptos? I started with tutorial example and the axum starter project.

I'll give [render_to_string](https://docs.rs/leptos/latest/leptos/ssr/fn.render_to_string.html) a try. However, I also want to avoid having to call render_to_string on every endpoint where I return html. May be I'll need to create some helper function that does it for me.

**Reply by gbj** [reply]
Not sure what you mean by configuring the app to just use the core leptos. 

I just ran
```
cargo init --bin leptos-ssr
cd leptos-ssr
cargo add leptos --features=ssr,experimental-islands
```
edit `src/main.rs`
```rust
use leptos::*;

fn main() {
    println!("{}", leptos::ssr::render_to_string(|| view! {
        <html>
            <body>
                <p>"Hello, world!"</p>
            </body>
        </html>
    }));
}
```
and then `cargo run`

*All* of the complexity in the templates and setup comes from the need to compile for two separate targets, support hydration, etc. If you explicitly don't want WASM it is just a crate you can add to any Rust project like anything else.

**Reply by nripendra** [reply]
Thanks! This is exactly what I'm looking for. Basically something to start from very scratch. I'm pretty new to rust.
